### PR TITLE
Compact & Generic records code samples API-1160

### PR DESCRIPTION
**Pull request by srknzl**
Includes code samples of compact and generic records

### CODE REVIEW

**Review by zpqrtbnk — DISMISSED** [review]
Approved.

File: code_samples/sql-data-types.js [hunk 1]
```diff
@@ -176,6 +203,49 @@ const objectExample = async (client, classId, factoryId) => {
...
+    );
+
+    const rowMetadata = result.rowMetadata;
+    const columnIndex = rowMetadata.findColumn('age');
+    const columnMetadata = rowMetadata.getColumn(columnIndex);
+    console.log(SqlColumnType[columnMetadata.type]); // INTEGER
+
+    const columnIndex2 = rowMetadata.findColumn('id');
+    const columnMetadata2 = rowMetadata.getColumn(columnIndex2);
+    console.log(SqlColumnType[columnMetadata2.type]); // BIGINT
```
Comment (yuce) [inline code_samples/sql-data-types.js:242]: Similar comment as above.

File: code_samples/compact_serialization_sample.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        return Employee;
+    }
+
+    getTypeName() {
+        return 'Employee';
+    }
+
+    read(reader) {
+        // Using getFieldKind method, you can use a default value if such a field does not exist.
+        const age = reader.getFieldKind('age') !== FieldKind.INT32 ? 0 : reader.readInt32('age');
```
[the remark this answers is not in the document]
Comment (srknzl) [reply code_samples/compact_serialization_sample.js:38]: `Reading a default value if a field is not found` use case is not advanced imo. And that's the only way of doing it. Do you think if we should create another sample just for this use case?

[the remark this answers is not in the document]
Comment (srknzl) [reply code_samples/compact_serialization_sample.js:38]: Added a new code sample.

**Review by yuce — APPROVED** [review]

**Review by zpqrtbnk — APPROVED** [review]
Approved.

**Pull request merged**